### PR TITLE
PI-8520 - Part II - handle "forget password logic"

### PR DESCRIPTION
**Pull request by awesselburg**

### CODE REVIEW

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php [hunk 1]
```diff
@@ -21,6 +21,16 @@
  */
 class Shopgate_Cloudapi_Model_Frontend_Observer_Layout
 {
+    /**
+     * Define custom handles
+     */
+    const CUSTOM_HANDLES = array(
```
Comment (augsteyer) [inline src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php]: I think php5.3/5.4 does not allow this kind of stuff, use property instead

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php [hunk 2]
```diff
@@ -31,9 +41,32 @@ public function execute(Varien_Event_Observer $observer)
...
+
+    /**
+     * @param Mage_Core_Model_Layout $layout
+     */
+    private function addCustomHandle($layout)
+    {
+        /** @var Mage_Core_Controller_Request_Http $request */
+        $request = Mage::app()->getRequest();
+        $path    = sprintf(
+            "/%s/%s/%s",
```
Comment (augsteyer) [inline src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php]: should be single quotes, I am fairly certain the %s should still work

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php [hunk 3]
```diff
@@ -21,6 +21,16 @@
...
+    const CUSTOM_HANDLES = array(
+        array(
+            'path'   => '/customer/account/login',
+            'handle' => 'shopgate_cloudapi_customer_account_login'
+        )
+    );
+
     /**
      * @param Varien_Event_Observer $observer
      * @return $this
```
Comment (augsteyer) [inline src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php]: ugh, in the previous I forgot to mention, should have a space between param and return. Should also declare the `Varien_Exception` throws

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php [hunk 4]
```diff
@@ -31,9 +41,32 @@ public function execute(Varien_Event_Observer $observer)
...
+        $layout->getUpdate()->addHandle('shopgate_cloudapi_default');
+        $this->addCustomHandle($layout);
 
         return $this;
     }
+
+    /**
+     * @param Mage_Core_Model_Layout $layout
+     */
+    private function addCustomHandle($layout)
```
Comment (augsteyer) [inline src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php]: name should be plural as you are adding more than one (possibly)

File: src/app/code/community/Shopgate/Cloudapi/etc/config.xml [hunk 5]
```diff
@@ -167,6 +167,8 @@
                         .page-title,
                         .customer-account-create .rewards,
                         .customer-account-create .back-link,
+                        .customer-account-forgotpassword .back-link,
+                        #login-form,
```
Comment (augsteyer) [inline src/app/code/community/Shopgate/Cloudapi/etc/config.xml]: should this have a page handle of some sort? I worry about these open css display non's ^_^

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php [hunk 6]
```diff
@@ -21,6 +21,16 @@
  */
 class Shopgate_Cloudapi_Model_Frontend_Observer_Layout
 {
+    /**
+     * Define custom handles
+     */
+    const CUSTOM_HANDLES = array(
+        array(
+            'path'   => '/customer/account/login',
+            'handle' => 'shopgate_cloudapi_customer_account_login'
```
Comment (augsteyer) [inline src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php]: you also want to add the other handle ```shopgate_cloudapi_customer_customer_account_create``` ?

**Comment by augsteyer** [comment]
also, don't forget the changelog. I fear we may have forgotten it quite a few PR's now.

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php (hunk 6 above)
Comment (awesselburg) [reply src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php]: 'shopgate_cloudapi_customer_customer_account_create' is our own template / route. We only need the custom handle for default routes to inject JS or add / remove blocks. I'm sure that we need in the future other handles, but currently its only 'shopgate_cloudapi_customer_account_login' (customer_account_login)

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (awesselburg) [reply src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php]: done

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php (hunk 1 above)
Comment (awesselburg) [reply src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php]: fixed

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php (hunk 2 above)
Comment (awesselburg) [reply src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php]: changed

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php (hunk 3 above)
Comment (awesselburg) [reply src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php]: changed

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php (hunk 4 above)
Comment (awesselburg) [reply src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php]: done

File: src/app/code/community/Shopgate/Cloudapi/etc/config.xml (hunk 5 above)
Comment (awesselburg) [reply src/app/code/community/Shopgate/Cloudapi/etc/config.xml]: extended to feel save ^^

File: src/app/design/frontend/base/default/template/shopgate/cloudapi/content/customer/account/login.phtml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+<div class="back-to-login">
+    <button type="button" class="button"
+            title="<?php echo Mage::helper('shopgate_cloudapi')->quoteEscape($this->__('Back to login')) ?>" name="send" id="send2">
+        <span><span><?php echo $this->__('Back to login') ?></span></span></button>
+</div>
```
Comment (augsteyer) [inline src/app/design/frontend/base/default/template/shopgate/cloudapi/content/customer/account/login.phtml]: this seems to be not going the right direction with the template locations, can you change the locations of these templates while we are in these early stages? e.g.
```
frontend/base/default/template/shopgate/cloudapi/customer/account/login/header.phtml
frontend/base/default/template/shopgate/cloudapi/customer/account/login/content.phtml
```
Adjust `modman` if necessary.

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php (hunk 6 above)
Comment (augsteyer) [reply src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/Layout.php]: oh yeah, oops :)

File: src/app/design/frontend/base/default/template/shopgate/cloudapi/content/customer/account/login.phtml (hunk 7 above)
Comment (awesselburg) [reply src/app/design/frontend/base/default/template/shopgate/cloudapi/content/customer/account/login.phtml]: you are right, looks cleaner now. pls. check.

**Pull request merged**